### DISCUSSION TITLE
RadioLib Initialization failed code-2

### DISCUSSION THREAD

**Discussion by veronal** [timmbogner/Farm-Data-Relay-System#173] [General]
First post with the group, have been working with the FDRS system for a couple of months now, thoroughly impressed, kudos to Timm,
Am getting the above error with the LilyGo v1 board (https://www.lilygo.cc/products/lora32-v1-0-lora-868mhz-915mhz).
The thing is it was working before, and since then a couple of things have happened, I experience a file lose which I now think I have recovered from, and I upgraded my library to the latest version.  I have checked settings in the config file and that seem to be correct so I am at a loss.. has anybody else experience the same ?  I have attached my config file ...  thanks
`//  FARM DATA RELAY SYSTEM
//
//  GATEWAY CONFIGURATION

//Addresses
#define UNIT_MAC           0x02  // The address of this gateway

#define ESPNOW_NEIGHBOR_1  0x00  // Address of ESP-NOW neighbor #1
#define ESPNOW_NEIGHBOR_2  0x00  // Address of ESP-NOW neighbor #2
#define LORA_NEIGHBOR_1    0x01  // Address of LoRa neighbor #1
#define LORA_NEIGHBOR_2    0x00  // Address of LoRa neighbor #2

// Interfaces
#define USE_ESPNOW  
#define USE_LORA
//#define USE_WIFI  // Will cause errors if used with ESP-NOW. Use a serial link instead!
//#define USE_ETHERNET

// Routing
// Options: sendESPNowNbr(1 or 2); sendESPNowPeers(); sendLoRaNbr(1 or 2); broadcastLoRa(); sendSerial(); sendMQTT();
#define ESPNOWG_ACT    sendLoRaNbr(1);
#define LORAG_ACT      sendLoRaNbr(1);
#define SERIAL_ACT     
#define MQTT_ACT          
#define INTERNAL_ACT   sendLoRaNbr(1);
#define ESPNOW1_ACT    
#define ESPNOW2_ACT                    
#define LORA1_ACT      sendESPNowPeers(); //broadcastLoRa(); //sendLoRaNbr(2);
#define LORA2_ACT      //sendLoRaNbr(1);

// LoRa Configuration
#define RADIOLIB_MODULE SX1276
#define LORA_SS    18
#define LORA_RST   14
#define LORA_DIO   26
#define LORA_BUSY  33
//#define USE_SX126X

#define LORA_TXPWR 17   // LoRa TX power in dBm (: +2dBm - +17dBm (for SX1276-7) +20dBm (for SX1278))

//#define CUSTOM_SPI
#define LORA_SPI_SCK  5
#define LORA_SPI_MISO 19
#define LORA_SPI_MOSI 27

#define LORA_ACK

#define FDRS_DEBUG     // Enable USB-Serial debugging

// OLED -- Displays console debugging messages on an SSD1306 I²C OLED
#define USE_OLED    
#define OLED_HEADER "FDRS"
#define OLED_PAGE_SECS 30
#define OLED_SDA 4
#define OLED_SCL 15
#define OLED_RST 16

// UART data interface pins (if available)
#define RXD2 14
#define TXD2 15

//#define USE_LR  // Use ESP-NOW LR mode (ESP32 only)

// WiFi and MQTT Credentials  -- These will override the global settings
//#define WIFI_SSID   "Your SSID"  
//#define WIFI_PASS   "Your Password"

//#define MQTT_ADDR   "192.168.0.8"
//#define MQTT_PORT   1883 // Default MQTT port is 1883
//#define MQTT_AUTH   //Enable MQTT authentication 
//#define MQTT_USER   "Your MQTT Username"
//#define MQTT_PASS   "Your MQTT Password"`

**Comment by aviateur17** [comment]
Could be something wrong with the pin assignments.  Try setting LORA_BUSY to RADIOLIB_NC and changing TXD2 and RXD2 to some unused pins like 32 and 33.  Play with the Pin assignments a bit.
```
// FARM DATA RELAY SYSTEM
//
// GATEWAY CONFIGURATION

//Addresses
#define UNIT_MAC 0x02 // The address of this gateway

#define ESPNOW_NEIGHBOR_1 0x00 // Address of ESP-NOW neighbor https://github.com/timmbogner/Farm-Data-Relay-System/issues/1
#define ESPNOW_NEIGHBOR_2 0x00 // Address of ESP-NOW neighbor https://github.com/timmbogner/Farm-Data-Relay-System/issues/2
#define LORA_NEIGHBOR_1 0x01 // Address of LoRa neighbor https://github.com/timmbogner/Farm-Data-Relay-System/issues/1
#define LORA_NEIGHBOR_2 0x00 // Address of LoRa neighbor https://github.com/timmbogner/Farm-Data-Relay-System/issues/2

// Interfaces
#define USE_ESPNOW
#define USE_LORA
//#define USE_WIFI // Will cause errors if used with ESP-NOW. Use a serial link instead!
//#define USE_ETHERNET

// Routing
// Options: sendESPNowNbr(1 or 2); sendESPNowPeers(); sendLoRaNbr(1 or 2); broadcastLoRa(); sendSerial(); sendMQTT();
#define ESPNOWG_ACT sendLoRaNbr(1);
#define LORAG_ACT sendLoRaNbr(1);
#define SERIAL_ACT
#define MQTT_ACT
#define INTERNAL_ACT sendLoRaNbr(1);
#define ESPNOW1_ACT
#define ESPNOW2_ACT
#define LORA1_ACT sendESPNowPeers(); //broadcastLoRa(); //sendLoRaNbr(2);
#define LORA2_ACT //sendLoRaNbr(1);

// LoRa Configuration
#define RADIOLIB_MODULE SX1276
#define LORA_SS 18
#define LORA_RST 14
#define LORA_DIO 26
#define LORA_BUSY RADIOLIB_NC
//#define USE_SX126X

#define LORA_TXPWR 17 // LoRa TX power in dBm (: +2dBm - +17dBm (for SX1276-7) +20dBm (for SX1278))

//#define CUSTOM_SPI
#define LORA_SPI_SCK 5
#define LORA_SPI_MISO 19
#define LORA_SPI_MOSI 27

#define LORA_ACK

#define FDRS_DEBUG // Enable USB-Serial debugging

// OLED -- Displays console debugging messages on an SSD1306 I²C OLED
#define USE_OLED
#define OLED_HEADER "FDRS"
#define OLED_PAGE_SECS 30
#define OLED_SDA 4
#define OLED_SCL 15
#define OLED_RST 16

// UART data interface pins (if available)
#define RXD2 32
#define TXD2 33

//#define USE_LR // Use ESP-NOW LR mode (ESP32 only)

// WiFi and MQTT Credentials -- These will override the global settings
//#define WIFI_SSID "Your SSID"
//#define WIFI_PASS "Your Password"

//#define MQTT_ADDR "192.168.0.8"
//#define MQTT_PORT 1883 // Default MQTT port is 1883
//#define MQTT_AUTH //Enable MQTT authentication
//#define MQTT_USER "Your MQTT Username"
//#define MQTT_PASS "Your MQTT Password"
```

**Comment by aviateur17** [comment]
You also may try uncommenting the CUSTOM_SPI line 
```
#define CUSTOM_SPI

**Reply by veronal** [reply]
Success!! made those modifications and I no longer get the error
thanks, would not thought of that myself...
I understand the LORA_BUSY change, but not clear on what the  CUSTOM_SPI change does  ??
Can you please elaborate ?  thanks again

**Reply by timmbogner** [reply]
Hey @veronal,
I think Jeff must have been ahead of me on this, because it just dawned on me what the likely problem is/was:

If the wrong board is selected, RadioLib sometimes has an issue starting up. Have you selected "TTGO-LoRa32-OLED" as your board?
